### PR TITLE
fix: update database layout in document

### PR DESCRIPTION
**Pull request by appflowy**
#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by appflowy** [comment]
fix #2748

**Comment by richardshiue** [comment]
LGTM, could we also use 'editor/calendar' in `appflowy_flutter/lib/plugins/database_view/calendar/calendar.dart`?

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart [hunk 1]
```diff
@@ -7,6 +7,10 @@ import 'package:appflowy_editor/appflowy_editor.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
+import 'editor_plugins/calendar/calendar_node_widget.dart';
+import 'editor_plugins/calendar/calendar_view_menu_item.dart';
+import 'editor_plugins/calendar/calendar_menu_item.dart';
```
Comment (richardshiue) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart:12]: Perhaps we could export these three files in `appflowy_flutter/lib/plugins/document/presentation/editor_plugins/plugins.dart` like the other database views? That way we only need to import that file. Totally okay to do it this way though

**Review by richardshiue — APPROVED** [review]

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          (r) => null,
+        );
+      },
+    );
+
+    _layoutTypeChangeNotifier = ViewLayoutTypeChangeNotifier(
+      layoutType: widget.view.layout,
+    );
+
+    super.initState();
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:50]: Best practice to have the super.initState() first.

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  /// The view will be updated by the [ViewListener].
+  late ViewPB view;
+
+  @override
+  void initState() {
+    view = widget.view;
+    _listener = ViewListener(view: widget.view);
+    _listener.start(
+      onViewUpdated: (result) {
+        result.fold(
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:34]: Can't we use `foldOrNull` here

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/calendar/calendar_menu_item.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  ),
+  keywords: ['referenced', 'calendar', 'database'],
+  handler: (editorState, menuService, context) {
+    final container = Overlay.of(context);
+    showLinkToPageMenu(
+      container,
+      editorState,
+      menuService,
+      ViewLayoutPB.Calendar,
+    );
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/calendar/calendar_menu_item.dart:24]:
```suggestion
    showLinkToPageMenu(
      Overlay.of(context),
      editorState,
      menuService,
      ViewLayoutPB.Calendar,
    );
```

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart (hunk 2 above)
Comment (appflowy) [reply frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:50]: Got it.

File: frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart (hunk 1 above)
Comment (appflowy) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/editor_page.dart:12]: Sure.

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart (hunk 3 above)
Comment (appflowy) [reply frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:34]: Just follow the old pattern. I don't know there is foldOrNull. Will try it later

Comment (LucasXu0) [reply frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:34]: there's no `foldOrNull`.

Comment (appflowy) [reply frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:34]: Oh. I forgot to print the error. So keep using fold

**Comment by appflowy** [comment]
> LGTM, could we also use 'editor/calendar' in `appflowy_flutter/lib/plugins/database_view/calendar/calendar.dart`?

yep

**Review by LucasXu0 — APPROVED** [review]

File: frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart [hunk 5]
```diff
@@ -59,30 +44,32 @@ class _DatabaseViewWidgetState extends State<DatabaseViewWidget> {
...
+      ..start(
+        onViewUpdated: (result) {
+          result.fold(
+            (updatedView) {
+              if (mounted) {
+                view = updatedView;
+                _layoutTypeChangeNotifier.value = view.layout;
+              }
+            },
+            (r) => null,
```
Comment (appflowy) [inline frontend/appflowy_flutter/lib/plugins/database_view/widgets/database_view_widget.dart:68]: Log the error

**Pull request merged**